py-polars/polars/dataframe/frame.py

```diff
@@ -4966,6 +4976,12 @@ def groupby_dynamic(
             - 'tuesday': Start the window on the tuesday before the first data point.
             - ...
             - 'sunday': Start the window on the sunday before the first data point.
+        check_sorted
+            When the ``by`` argument is given, polars can not check sortedness
+            by the metadata and has to do a full scan on the index column to
+            verify data is sorted. This is expensive. If you are sure the
+            data within the by groups is sorted, you can set this to ``False``.
+            Doing so incorrectly will lead to incorrect output
```

Comment:
I might be misreading/misunderstanding, but from

> When the ``by`` argument is given, polars can not check sortedness by the metadata

I think it sounds like that if `by` isn't given, then polars checks for sortedness using the metadata.

It doesn't error in that case though?
```python
df = pl.DataFrame(
    {
        "idx": [1, 2, -1, 2, 1, 1],
        "group": [1, 1, 1, 2, 2, 1],
    }
)
print(df['idx'].flags)  # {'SORTED_ASC': False, 'SORTED_DESC': False}
print(df.groupby_dynamic("idx", every="2i").agg(pl.col("idx").alias("idx1")))  # no error
```

It would fail if `by=` was passed. Is it right that the `by=` case is stricter than the `by=None` one?

Reply:
It should error in the other case as well. It should check the flags. Thanks, will add test for those.